kore/src/main/java/org/kframework/POSet.java

```diff
@@ -186,6 +186,29 @@ public Optional<T> maximum() {
     return maximumLazy.get();
   }
 
+  private Map<T, Integer> computeConnectedComponents() {
```

Comment:
Please add a Javadoc comment that promptly describes the algorithm and why the greedy approach works (`directRelations` is transitively closed, what the loop ensures on `data`, etc).